crates/bevy_transform/src/components/transform2d.rs

```diff
@@ -0,0 +1,372 @@
+    #[inline]
+    pub fn up(&self) -> Vec2 {
+        self.local_y()
+    }
+
+    /// Equivalent to [`-local_y()`][Self::local_y]
+    #[inline]
+    pub fn down(&self) -> Vec2 {
+        -self.local_y()
+    }
+
+    /// Returns the rotation matrix from this transforms rotation.
+    #[inline]
+    pub fn rotation_matrix(&self) -> Mat2 {
+        Mat2::from_angle(self.rotation)
+    }
+
+    /// Computes the affine transformation matrix of this transform.
+    #[inline]
+    pub fn compute_matrix(&self) -> Mat3 {
+        Mat3::from_scale_angle_translation(self.scale, self.rotation, self.translation)
+    }
+
+    /// Computes the affine transform of this transform.
+    #[inline]
+    pub fn compute_affine(&self) -> Affine2 {
+        Affine2::from_scale_angle_translation(self.scale, self.rotation, self.translation)
+    }
+
+    /// Translates this [`Transform2d`] around a `point` in space.
+    ///
+    /// If this [`Transform2d`] has a parent, the `point` is relative to the [`Transform2d`] of the parent.
+    #[inline]
+    pub fn translate_around(&mut self, point: Vec2, angle: f32) {
+        self.translation = point + Mat2::from_angle(angle) * (self.translation - point);
+    }
+
+    /// Rotates this [`Transform2d`] around a `point` in space.
+    ///
+    /// If this [`Transform2d`] has a parent, the `point` is relative to the [`Transform2d`] of the parent.
+    #[inline]
+    pub fn rotate_around(&mut self, point: Vec2, angle: f32) {
+        self.translate_around(point, angle);
+        self.rotation += angle;
+    }
+
+    /// Transforms the given `point`, applying scale, rotation and translation.
+    /// `z_translation` is ignored.
+    ///
+    /// If this [`Transform2d`] has a parent, this will transform a `point` that is
+    /// relative to the parent's [`Transform2d`] into one relative to this [`Transform2d`].
+    ///
+    /// If this [`Transform2d`] does not have a parent, this will transform a `point`
+    /// that is in global space into one relative to this [`Transform2d`].
+    ///
+    /// If you want to transform a `point` in global space to the local space of this [`Transform2d`],
+    /// consider using [`GlobalTransform2d::transform_point()`](super::GlobalTransform2d::transform_point) instead.
+    #[inline]
+    pub fn transform_point(&self, mut point: Vec2) -> Vec2 {
+        point *= self.scale;
+        point = self.rotation_matrix() * point;
+        point += self.translation;
+        point
+    }
+
+    /// Multiplies `self` with `transform` component by component, returning the
+    /// resulting [`Transform2d`]
+    #[inline]
+    #[must_use]
+    pub fn mul_transform(&self, transform: Transform2d) -> Self {
+        let translation = self.transform_point(transform.translation);
+        let rotation = self.rotation + transform.rotation;
+        let scale = self.scale * transform.scale;
+        let z_translation = self.z_translation + transform.z_translation;
```

Comment:
adding the z's here feels very strange to me. I feel like in most instances if you multiply by another transform you'd want to leave the z unchanged. I think I'd prefer the user space z to be a separate component.

Reply:
A better API for z-sorting is its own can of worms that I would rather punt to a future PR.
Current behavior on this PR is identical to main.

Reply:
a separate z component can have the same z sorting behavior as on main.